Foundation/Random.hs

```diff
@@ -37,13 +38,23 @@ import qualified Foundation.Array.Unboxed as A
 import qualified Foundation.Array.Unboxed.Mutable as A
 import           GHC.ST
 import qualified Prelude
+import qualified Foreign.Marshal.Alloc (alloca)
 
 -- | A monad constraint that allows to generate random bytes
 class (Functor m, Applicative m, Monad m) => MonadRandom m where
     getRandomBytes :: CountOf Word8 -> m (UArray Word8)
+    getRandomWord64 :: m Word64
+    getRandomF32 :: m Float
```

Comment:
I'm tempted to just make this just a downsize of `Double` (calling the doubleToFloat function that is available under some name I can't remember from ghc prim) instead of making it part of the class